pkg/controllers/work/apply_controller.go

```diff
@@ -340,6 +346,16 @@ func (r *ApplyWorkReconciler) applyUnstructured(ctx context.Context, gvr schema.
 		Name:      manifestObj.GetName(),
 		Namespace: manifestObj.GetNamespace(),
 	}
+	// Marshaling adds about 100-200 bytes to the object's actual size
+	marshaledObj, marshalErr := json.Marshal(manifestObj)
+	if marshalErr != nil {
+		return nil, ManifestNoChangeAction, marshalErr
+	}
+	if len(marshaledObj) > (TotalAnnotationSizeLimitB - len(manifestHashAnnotation) - len(lastAppliedConfigAnnotation) - manifestHashLength) {
```

Comment:
Why are we checking the size of the manifestObj instead of the size of the last applied Annotation?

Reply:
does the object have the lastAppliedConfigAnnotation label now?

Reply:
Computing and checking the last applied config, current implementation doesn't have the label hence I'm subtracting it from the 256KB allowed for annotation limit